diffsmhm/tests/test_loader.py

```diff
@@ -48,3 +69,235 @@ def test_munge_halos():
     assert np.array_equal(halos["halo_x"], orig_halos["x"])
     assert np.array_equal(halos["halo_y"], orig_halos["y"])
     assert np.array_equal(halos["halo_z"], orig_halos["z"])
+
+
+@pytest.mark.mpi_skip
+def test_find_and_write_most_massive_hosts_tree_walk():
+    if RANK == 0:
+        make_test_catalogs(1)
+    COMM.Barrier()
+
+    script_dir = os.path.realpath(os.path.dirname(__file__))
+    testfile = script_dir+"/testing_catalogs/mock_halos_1.h5"
+    mmhid, mmh_x, mmh_y, mmh_z, mmh_dist = find_and_write_most_massive_hosts(
+                                                testfile, export=True
+                                           )
+
+    expected_mmhid = [-1, 1, 1, 1]
+
+    expected_mmh_x = [10, 10, 10, 10]
+    expected_mmh_y = [-10, -10, -10, -10]
+    expected_mmh_z = [10, 10, 10, 10]
+
+    if RANK == 0:
+        with h5py.File(testfile, "r") as f:
+            expected_mmh_dist = np.sqrt(np.power(f["x"][...] - expected_mmh_x, 2) +
+                                        np.power(f["y"][...] - expected_mmh_y, 2) +
+                                        np.power(f["z"][...] - expected_mmh_z, 2)
+                                        )
+
+        ok = True
+
+        try:
+            assert np.allclose(expected_mmhid, mmhid)
+
+            assert np.allclose(expected_mmh_x, mmh_x)
+            assert np.allclose(expected_mmh_y, mmh_y)
+            assert np.allclose(expected_mmh_z, mmh_z)
+
+            assert np.allclose(expected_mmh_dist, mmh_dist)
+
+            f = h5py.File(testfile, "r")
+            assert np.allclose(f["mmhid"], expected_mmhid)
+            assert np.allclose(f["mmh_x"], expected_mmh_x)
+            assert np.allclose(f["mmh_y"], expected_mmh_y)
+            assert np.allclose(f["mmh_z"], expected_mmh_z)
+            assert np.allclose(f["mmh_dist"], expected_mmh_dist)
+            f.close()
```

Comment:
You should use a context manager here or a finally clause to ensure the file `f` is closed properly.